pandas/_libs/hashtable.pyx

```diff
@@ -64,6 +64,12 @@ cdef class Factorizer:
         >>> factorize(np.array([1,2,np.nan], dtype='O'), na_sentinel=20)
         array([ 0,  1, 20])
         """
+        if self.uniques.external_view_exists:
+            uniques = ObjectVector()
+            data = self.uniques.to_array()
+            for v in data:
+                uniques.append(v)
```

Comment:
for consistency maybe we should add ``.extend`` to String/Object Vectors. can you add?